diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/StringTemplateFormatRule.kt

```diff
@@ -0,0 +1,70 @@
+import com.pinterest.ktlint.core.ast.ElementType.IDENTIFIER
+import com.pinterest.ktlint.core.ast.ElementType.LONG_STRING_TEMPLATE_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.REFERENCE_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.SHORT_STRING_TEMPLATE_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.SHORT_TEMPLATE_ENTRY_START
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithSpecificType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+
+/**
+ * In String templates there should not be redundant curly braces. In case of using a not complex statement (one argument)
+ * there should not be curly braces.
+ */
+class StringTemplateFormatRule(private val configRules: List<RulesConfig>) : Rule("string-template-format") {
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        when (node.elementType) {
+            LONG_STRING_TEMPLATE_ENTRY -> handleLongStringTemplate(node)
+            SHORT_STRING_TEMPLATE_ENTRY -> handleShortStringTemplate(node)
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun handleLongStringTemplate(node: ASTNode) {
+        // Checking if in long templates {a.foo()} there are function calls or class toString call
+        if (node.findAllNodesWithSpecificType(COLONCOLON).isEmpty()
+                && node.findAllNodesWithSpecificType(DOT_QUALIFIED_EXPRESSION).isEmpty()) {
+            Warnings.STRING_TEMPLATE_CURLY_BRACES.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset, node) {
+                val identifierName = node.findChildByType(REFERENCE_EXPRESSION)!!.text
+                val shortTemplate = CompositeElement(SHORT_STRING_TEMPLATE_ENTRY)
+                val reference = CompositeElement(REFERENCE_EXPRESSION)
+
+                node.treeParent.addChild(shortTemplate, node)
+                shortTemplate.addChild(LeafPsiElement(SHORT_TEMPLATE_ENTRY_START, "$"), null)
+                shortTemplate.addChild(reference)
+                reference.addChild(LeafPsiElement(IDENTIFIER, identifierName))
+                node.treeParent.removeChild(node)
+            }
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun handleShortStringTemplate(node: ASTNode) {
+        val identifierName = node.findChildByType(REFERENCE_EXPRESSION)!!.text
```

Comment:
Here I can't think of an example where there is no reference, but to be safe please add handling of null.